scripts/portstat

```diff
@@ -156,6 +160,48 @@ class Portstat(object):
         else:
             print tabulate(table, header_all if print_all else header, tablefmt='simple', stralign='right')
 
+    def network_errors_cnstat_print(self, cnstat_dict, use_json, print_all):
```

Comment:
You don't need the "print_all" parameter here, which makes me think you may be able to decrease even more code duplication by removing these new functions and instead passing two more parameters ("errors_only", "rates_only") to the `cnstat_print(self, cnstat_dict, use_json, print_all)` function above. The -e, -R and -a flags will all be mutually exclusive. Then, you can simply add to the if/else statement, something like:

```
            if print_all:
                table.append((key, self.get_port_state(key),
                              data.rx_ok, STATUS_NA, STATUS_NA, STATUS_NA, data.rx_err,
                              data.rx_drop, data.rx_ovr,
                              data.tx_ok, STATUS_NA, STATUS_NA, STATUS_NA, data.tx_err,
                              data.tx_drop, data.tx_ovr))
            elif errors_only:
                table.append((key, self.get_port_state(key),
                              data.rx_err, data.rx_drop, data.rx_ovr,
                              data.tx_err, data.tx_drop, data.tx_ovr))
            elif rates_only:
                table.append((key, self.get_port_state(key),
                              data.rx_ok, STATUS_NA, STATUS_NA, STATUS_NA,
                              data.tx_ok, STATUS_NA, STATUS_NA, STATUS_NA))
            else:
                table.append((key, self.get_port_state(key),
                              data.rx_ok, STATUS_NA, STATUS_NA, data.rx_err,
                              data.rx_drop, data.rx_ovr,
                              data.tx_ok, STATUS_NA, STATUS_NA, data.tx_err,
                              data.tx_drop, data.tx_ovr))
```

And in each of these cases, you could also set the header you want to use.

Reply:
updated files.